### PR TITLE
PAYARA-3577 Build Payara Server & Payara Micro using JDK8 and get them to pass the Test Suite

### PR DESCRIPTION
**Pull request by jGauravGupta**
Depends on https://github.com/javaee-samples/javaee7-samples/pull/449/

### CODE REVIEW

File: appserver/admin/gf_template/src/main/resources/config/domain.xml [hunk 1]
```diff
@@ -261,6 +257,7 @@
         <jvm-options>[1.8.0u121|1.8.0u160]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.7.jar</jvm-options>
         <jvm-options>[1.8.0u161|1.8.0u190]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.jar</jvm-options>
         <jvm-options>[1.8.0u191|1.8.0u500]-Xbootclasspath/p:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap-1.8.1.jar</jvm-options>
+        <jvm-options>[9|]-Xbootclasspath/a:${com.sun.aas.installRoot}/lib/grizzly-npn-bootstrap.jar</jvm-options>
```
Comment (smillidge) [inline appserver/admin/gf_template/src/main/resources/config/domain.xml]: Why does JDK 11 need the npn-boot jar? Doesn't JDK 11 support NPN negotiation?

Comment (jGauravGupta) [reply appserver/admin/gf_template/src/main/resources/config/domain.xml]: [Http2AddOn](https://github.com/eclipse-ee4j/grizzly/blob/master/modules/http2/src/main/java/org/glassfish/grizzly/http2/Http2AddOn.java#L121) requires `grizzly-npn-bootstrap.jar#org.glassfish.grizzly.npn.AlpnServerNegotiator` class.

Comment (smillidge) [reply appserver/admin/gf_template/src/main/resources/config/domain.xml]: OK that may need Grizzly work then as Alpn negotiation I thought was in the SSL stack of the JDK now

Comment (smillidge) [reply appserver/admin/gf_template/src/main/resources/config/domain.xml]: probably not enough to stop this PR going in but can you raise an issue on Grizzly

Comment (jGauravGupta) [reply appserver/admin/gf_template/src/main/resources/config/domain.xml]: Added the `grizzly-npn-api.jar` to classpath for JDK 9+

**Pull request merged**